skorch/tests/callbacks/test_logging.py

```diff
@@ -10,7 +10,143 @@
+        X, y = X[:40], y[:40]
+        return X, y
+
+    @pytest.fixture
+    def neptune_logger_cls(self):
+        from skorch.callbacks import NeptuneLogger
+        return NeptuneLogger
+
+    @pytest.fixture
+    def neptune_experiment_cls(self):
+        import neptune
+        neptune.init(project_qualified_name="tests/dry-run",
+                     backend=neptune.OfflineBackend())
+        return neptune.create_experiment
+
+    @pytest.fixture
+    def mock_experiment(self, neptune_experiment_cls):
+        mock = Mock(spec=neptune_experiment_cls)
+        mock.log_metric = Mock()
+        mock.stop = Mock()
+        return mock
+
+    @pytest.fixture
+    def net_fitted(
+            self,
+            net_cls,
+            classifier_module,
+            data,
+            neptune_logger_cls,
+            mock_experiment,
+    ):
+        return net_cls(
+            classifier_module,
+            callbacks=[neptune_logger_cls(mock_experiment)],
+            max_epochs=3,
+        ).fit(*data)
+
+    def test_experiment_closed_automatically(self, net_fitted, mock_experiment):
+        assert mock_experiment.stop.call_count == 1
+
+    def test_experiment_not_closed(
+            self,
+            net_cls,
+            classifier_module,
+            data,
+            neptune_logger_cls,
+            mock_experiment,
+    ):
+        net_cls(
+            classifier_module,
+            callbacks=[
+                neptune_logger_cls(mock_experiment, close_after_train=False)],
+            max_epochs=2,
+        ).fit(*data)
+        assert mock_experiment.stop.call_count == 0
+
+    def test_ignore_keys(
+            self,
+            net_cls,
+            classifier_module,
+            data,
+            neptune_logger_cls,
+            mock_experiment,
+    ):
+        # ignore 'dur' and 'valid_loss', 'unknown' doesn't exist but
+        # this should not cause a problem
+        npt = neptune_logger_cls(
+            mock_experiment, keys_ignored=['dur', 'valid_loss', 'unknown'])
+        net_cls(
+            classifier_module,
+            callbacks=[npt],
+            max_epochs=3,
+        ).fit(*data)
+
+    def test_keys_ignored_is_string(self, neptune_logger_cls, mock_experiment):
+        npt = neptune_logger_cls(mock_experiment,
+                                 keys_ignored='a-key').initialize()
+        expected = {'a-key', 'batches'}
+        assert npt.keys_ignored_ == expected
+
+    def test_fit_with_real_experiment(
+            self,
+            net_cls,
+            classifier_module,
+            data,
+            neptune_logger_cls,
+            neptune_experiment_cls,
+    ):
+        net = net_cls(
+            classifier_module,
+            callbacks=[neptune_logger_cls(neptune_experiment_cls())],
+            max_epochs=5,
+        )
+        net.fit(*data)
+
+    def test_fit_with_dict_input(
+            self,
+            net_cls,
+            classifier_module,
```

Comment:
argument not used (probably copied from tensorboard test that also doesn't need it). Tbh, I think this whole test can be removed here.

Reply:
dropped it